src/components/DataVisualization/Table/Table.vue

```diff
@@ -0,0 +1,170 @@
+<template>
+  <div class="table-container">
+    <div
+      v-if="hasActionsBar"
+      :class="{ 'actions-bar': !isMobile, 'actions-bar-mobile': isMobile }"
+    >
+      <div class="search-input">
+        <PbSearchInput
+          v-model="state.searchTerm"
+          placeholder="O que você procura?"
+          @search="$emit('search', state.searchTerm)"
+          @clear-input="$emit('clear-input')"
+        />
+      </div>
+      
+      <slot name="extra-actions" />
+    </div>
+
+    <TableHeader
+      :header="header"
+      :ellipsis-on-overflow="ellipsisOnOverflow"
+      :has-action-column="hasActionColumn"
+      :has-sort="hasSort"
+      :actions-size="actionsSize"
+      @sort="activeSorting => state.activeSorting = activeSorting"
+    />
+
+    <PbLoadingBar v-if="loading" />
+
+    <TableRows
+      :header="header"
+      :rows="sortedRows"
+      :metadata="metadata"
+      :ellipsis-on-overflow="ellipsisOnOverflow"
+      :highlight-on-hover="highlightOnHover"
+      :has-action-column="hasActionColumn"
+      :max-height="calculatedMaxHeight"
+      :actions-size="actionsSize"
+    >
+      <template #actions="props">
+        <slot
+          name="actions"
+          :row="props.row"
+          :metadata="props.metadata"
+        />
+      </template>
+    </TableRows>
+  </div>
+</template>
+
+<script>
+import TableHeader from './TableHeader.vue';
+import TableRows from './TableRows.vue';
+import PbSearchInput from '../../FiltersAndSearch/SearchInput/SearchInput.vue';
+import PbLoadingBar from '../../Loadings/LoadingBar/LoadingBar.vue';
+
+export default {
+  name: 'PbTable',
+
+  components: {
+    TableHeader,
+    TableRows,
+    PbSearchInput,
+    PbLoadingBar,
+  },
+
+  props: {
+    header: { type: Array, default: () => [] },
+    rows: { type: Array, default: () => [] },
+    metadata: { type: Array, default: () => [] },
+    loading: { type: Boolean, default: false },
+    ellipsisOnOverflow: { type: Boolean, default: true },
+    highlightOnHover: { type: Boolean, default: true },
+    hasActionColumn: { type: Boolean, default: true },
```

Comment:
I prefer using `actionColumn` rather than `hasActionColumn`

Reply:
I prefer to use hasXXX or useXXX instead of just XXX in boolean variables...

Reply:
I prefer to use hasX because it shows what is going on, when we receive a property like 'sort' I imagine that it is a sort configuration and not that I will activate the sort. Same thing to the other properties.

Reply:
I disagree, but ok